folksonomy/api.py

```diff
@@ -554,6 +570,37 @@ async def keys_list(response: Response,
     return JSONResponse(status_code=200, content=out[0], headers={"x-pg-timing": timing})
 
 
+@app.get("/values/{k}")
+async def get_unique_values(response: Response,
+                            k: str,
+                            owner: str = '',
+                            user: User = Depends(get_current_user)):
+    """
+    Get the unique values of a given property and the corresponding number of products
+
+    - **k**: The property key to get unique values for
+    - **owner**: None or empty for public tags, or your own user_id
+
+    Returns a JSON table containing all the values and the corresponding number of products
+    """
+    check_owner_user(user, owner, allow_anonymous=True)
+    k, _ = sanitize_data(k, None)
+    cur, timing = await db.db_exec("""
+        SELECT json_agg(j.j)::json FROM(
+            SELECT json_build_object(
+                'v', v,
+                'product_count', count(*)
+                ) as j
+            FROM folksonomy
+            WHERE owner=%s and k=%s
+            GROUP BY v) as j;
+        """,
```

Comment:
The formating does not help reading.

Moreover I think we should:
1. order by product_count desc
2. limit the result to a reasonable amount (100 ?) this could be a parameter

```suggestion
        SELECT json_agg(j.j)::json
        FROM (
            SELECT json_build_object(
                'v', v,
                'product_count', count(*)
                ) as j
            FROM folksonomy
            WHERE owner=%s and k=%s
            GROUP BY v
            ORDER BY count(*)
            LIMIT 100
        ) as j;
        """,
```

Reply:
Ok. Should `LIMIT 100` be a default, or should it be a positive parameter (no parameter = not limit)?
I think this should not be a default, but only a parameter.

And I also think we should add a parameter for a filter.

What would fit best?

1. /values/{key}?limit=50&offset=50&q=kca
2. /values/{key}?page_size=100&page=2&q=kca
3. or simply /values/{key}?limit=50&q=kca

I guess I would prefer solution 3.

Reply:
1. I really really want 100 to be a default @CharlesNepote (we have enough perf problem to avoid getting new one, because people don't read the docs)
2. I really want to a have a max for this limit (I would suggest 1000) again to avoid perfs pittfall (people using 1000000 as parameter and putting our infrastructure down)
3. +1 for solution 3 for now